buildpacks/java/tests/detect_test.go

```diff
@@ -94,7 +89,7 @@ func testDetect(t *testing.T, when spec.G, it spec.S) {
 
 		when("BP_FUNCTION is not configured", func() {
 			it.Before(func() {
-				Expect(os.Unsetenv("BP_FUNCTION")).To(Succeed())
+				t.Setenv("BP_FUNCTION", "")
```

Comment:
I think this may be the wrong context, BP_FUNCTION is configured but empty. By doing this here, we no longer cover the case where func.yaml does not exist and BP_FUNCTION is not configured

Reply:
That's a valid point. I'll move the "empty" case to the "misconfigured" area of the tests, and I'll change this to explicitly unset (using [this technique](https://github.com/golang/go/issues/52817#issuecomment-1131339120))